pipeline_steps/common.groovy

```diff
@@ -30,12 +30,25 @@ def void create_workspace_venv(){
     # UG-613 change TMPDIR to directory with more space
     export TMPDIR="/var/lib/jenkins/tmp"
 
+    # If the pip version we're using is not the same as the constraint then replace it
+    PIP_TARGET="\$(awk -F= '/^pip==/ {print \$3}' rpc-gating/constraints.txt)"
+    VENV_PYTHON=".venv/bin/python"
+    VENV_PIP=".venv/bin/pip"
+    if [[ "\$(\${VENV_PIP} --version)" != "pip \${PIP_TARGET}"* ]]; then
+      # Install a known version of pip, setuptools, and wheel in the venv
+      CURL_CMD="curl --silent --show-error --retry 5"
+      OUTPUT_FILE="get-pip.py"
+      \${CURL_CMD} https://bootstrap.pypa.io/get-pip.py > \${OUTPUT_FILE} \
+        || \${CURL_CMD} https://raw.githubusercontent.com/pypa/get-pip/master/get-pip.py > \${OUTPUT_FILE}
+      GETPIP_OPTIONS="pip setuptools wheel --constraint rpc-gating/constraints.txt"
+      \${VENV_PYTHON} \${OUTPUT_FILE} \${GETPIP_OPTIONS} \
```

Comment:
No need - when using a constraint it will automatically make sure that the constraint is installed.

```
root@trusty1:~# pip install virtualenv==15.0
Collecting virtualenv==15.0
Downloading virtualenv-15.0.0-py2.py3-none-any.whl (1.8MB)
    100% |████████████████████████████████| 1.8MB 652kB/s
Installing collected packages: virtualenv
  Found existing installation: virtualenv 15.1.0
    Uninstalling virtualenv-15.1.0:
      Successfully uninstalled virtualenv-15.1.0
Successfully installed virtualenv-15.0.0
root@trusty1:~# echo 'virtualenv==15.1.0' > constraint.txt
root@trusty1:~# pip install virtualenv --constraint constraint.txt
Collecting virtualenv==15.1.0 (from -c constraint.txt (line 1))
  Using cached virtualenv-15.1.0-py2.py3-none-any.whl
Installing collected packages: virtualenv
  Found existing installation: virtualenv 15.0.0
    Uninstalling virtualenv-15.0.0:
      Successfully uninstalled virtualenv-15.0.0
Successfully installed virtualenv-15.1.0
```